modules/plottinggl/src/plotters/scatterplotgl.cpp

```diff
@@ -299,30 +378,48 @@ void ScatterPlotGL::plot(const size2_t& dims, IndexBuffer* indexBuffer, bool use
     glDrawElements(GL_POINTS, static_cast<uint32_t>(indices->getSize()), indicesGL->getFormatType(),
                    nullptr);
     indicesGL->getBufferObject()->unbind();
-
     // draw selected and hovered points on top
     if (!selectedIndices_.empty()) {
         shader_.setUniform("has_color", 0);
         shader_.setUniform("default_color", properties_.selectionColor_.get());
-
-        std::vector<uint32_t> selected;
-        selected.reserve(selectedIndices_.size());
-        std::transform(selectedIndices_.begin(), selectedIndices_.end(),
-                       std::back_inserter(selected),
-                       [](size_t i) { return static_cast<uint32_t>(i); });
-
-        glDrawElements(GL_POINTS, static_cast<uint32_t>(selected.size()), GL_UNSIGNED_INT,
-                       selected.data());
+        if (selectedIndicesGLDirty_) {
+            std::vector<uint32_t> selected;
+            selected.reserve(selectedIndices_.size());
+#include <warn/push>
+#include <warn/ignore/conversion>  // We need to convert size_t to uint32
+            std::copy_if(
+                selectedIndices_.begin(), selectedIndices_.end(), std::back_inserter(selected),
+                [this](auto val) { return filteredIndices_.find(val) == filteredIndices_.end(); });
+#include <warn/pop>
+
+            if (selectedIndicesGL_.getSize() <
```

Comment:
This name is very bad... should be getByteSize or something very ugly like that....